flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalogFactory.java

```diff
@@ -41,13 +40,21 @@ public String identifier() {
     public Catalog create(String warehouse, Configuration options) {
         String uri =
                 Preconditions.checkNotNull(
-                        options.get(URI),
-                        URI.key() + " must be set for table store " + IDENTIFIER + " catalog");
+                        options.get(CatalogOptions.URI),
+                        CatalogOptions.URI.key()
+                                + " must be set for table store "
+                                + IDENTIFIER
+                                + " catalog");
+
         org.apache.hadoop.conf.Configuration hadoopConfig =
                 new org.apache.hadoop.conf.Configuration();
         options.toMap().forEach(hadoopConfig::set);
+        hadoopConfig.set(
```

Comment:
We can add test for here. May a mock `IMetaStoreClient` just implement `getAllDatabases` to verify.